client/helpers_test.go

```diff
@@ -662,136 +662,8 @@ func TestApplyTargetsChangeInvalidType(t *testing.T) {
 	assert.Error(t, err)
 }
 
-// A delegated role MUST NOT have both Paths and PathHashPrefixes defined.
-// These next 2 tests check that attempting to edit an existing role to
-// create an invalid role errors in both possible combinations.
-func TestApplyTargetsDelegationConflictPathsPrefixes(t *testing.T) {
-	_, repo, cs, err := testutils.EmptyRepo("docker.com/notary")
-	assert.NoError(t, err)
-
-	newKey, err := cs.Create("targets/level1", data.ED25519Key)
-	assert.NoError(t, err)
-
-	// create delegation
-	kl := data.KeyList{newKey}
-	td := &changelist.TufDelegation{
-		NewThreshold: 1,
-		AddKeys:      kl,
-		AddPaths:     []string{"level1"},
-	}
-
-	tdJSON, err := json.Marshal(td)
-	assert.NoError(t, err)
-
-	ch := changelist.NewTufChange(
-		changelist.ActionCreate,
-		"targets/level1",
-		changelist.TypeTargetsDelegation,
-		"",
-		tdJSON,
-	)
-
-	err = applyTargetsChange(repo, ch)
-	assert.NoError(t, err)
-
-	// add prefixes and update
-	td.AddPathHashPrefixes = []string{"abc"}
-
-	tdJSON, err = json.Marshal(td)
-	assert.NoError(t, err)
-
-	ch = changelist.NewTufChange(
-		changelist.ActionUpdate,
-		"targets/level1",
-		changelist.TypeTargetsDelegation,
-		"",
-		tdJSON,
-	)
-
-	err = applyTargetsChange(repo, ch)
-	assert.Error(t, err)
-}
-
-func TestApplyTargetsDelegationConflictPrefixesPaths(t *testing.T) {
-	_, repo, cs, err := testutils.EmptyRepo("docker.com/notary")
-	assert.NoError(t, err)
-
-	newKey, err := cs.Create("targets/level1", data.ED25519Key)
-	assert.NoError(t, err)
-
-	// create delegation
-	kl := data.KeyList{newKey}
-	td := &changelist.TufDelegation{
-		NewThreshold:        1,
-		AddKeys:             kl,
-		AddPathHashPrefixes: []string{"abc"},
-	}
-
-	tdJSON, err := json.Marshal(td)
-	assert.NoError(t, err)
-
-	ch := changelist.NewTufChange(
-		changelist.ActionCreate,
-		"targets/level1",
-		changelist.TypeTargetsDelegation,
-		"",
-		tdJSON,
-	)
-
-	err = applyTargetsChange(repo, ch)
-	assert.NoError(t, err)
-
-	// add paths and update
-	td.AddPaths = []string{"level1"}
-
-	tdJSON, err = json.Marshal(td)
-	assert.NoError(t, err)
-
-	ch = changelist.NewTufChange(
-		changelist.ActionUpdate,
-		"targets/level1",
-		changelist.TypeTargetsDelegation,
-		"",
-		tdJSON,
-	)
-
-	err = applyTargetsChange(repo, ch)
-	assert.Error(t, err)
-}
-
-func TestApplyTargetsDelegationCreateInvalid(t *testing.T) {
```

Comment:
Test is no longer relevant as it tests creating a role that has both paths and hash prefixes.